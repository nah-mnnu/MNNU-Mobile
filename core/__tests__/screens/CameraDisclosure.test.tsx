import { render, fireEvent } from '@testing-library/react-native'
import React from 'react'

import CameraDisclosure from '../../App/screens/CameraDisclosure'
import { testIdWithKey } from '../../App/utils/testable'

jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})

describe('CameraDisclosure Screen', () => {
  test('Renders correctly', () => {
    const cb = jest.fn()
    const tree = render(<CameraDisclosure didDismissCameraDisclosure={cb} />)

    expect(tree).toMatchSnapshot()
  })

  test('Okay button triggers callback', () => {
    const cb = jest.fn()
    const { getByTestId } = render(<CameraDisclosure didDismissCameraDisclosure={cb} />)

    const okayButton = getByTestId(testIdWithKey('Okay'))
    fireEvent(okayButton, 'press')

    expect(cb).toBeCalledTimes(1)
  })
})
