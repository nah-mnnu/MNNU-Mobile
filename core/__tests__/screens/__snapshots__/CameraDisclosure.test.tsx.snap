// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CameraDisclosure Screen Renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Array [
      Object {
        "backgroundColor": "#42803E",
        "flexGrow": 1,
        "paddingHorizontal": 25,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#313132",
        "borderColor": "#0099FF",
        "borderRadius": 5,
        "borderWidth": 1,
        "minWidth": 700,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 5,
          "paddingTop": 5,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
              "marginRight": 10,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#0099FF",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î¢Ž
        </Text>
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFFFFF",
            "flexShrink": 1,
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
        testID="com.ariesbifold:id/HeaderText"
      >
        PrivacyPolicy.Title
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "marginLeft": 40,
          "paddingBottom": 5,
          "paddingHorizontal": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "flexShrink": 1,
            "fontSize": 18,
            "fontWeight": "normal",
            "marginVertical": 16,
          }
        }
        testID="com.ariesbifold:id/BodyText"
      >
        PrivacyPolicy.CameraDisclosure
      </Text>
      <View
        accessibilityLabel="Global.Okay"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#42803E",
            "borderRadius": 4,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/Okay"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
              ]
            }
          >
            Global.Okay
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
